internal/config/image/builder.go

```diff
@@ -100,3 +105,57 @@ func WithMounts(mounts []specs.Mount) Option {
 		return nil
 	}
 }
+
+// WithSpec sets the OCI spec to use when creating the CUDA image.
+func WithSpec(s *specs.Spec) Option {
+	return func(b *builder) error {
+		b.spec = s
+		return nil
+	}
+}
+
+// WithSwarmResource sets the swarm resource for the CUDA image.
+func WithSwarmResource(resource ...string) Option {
+	return func(b *builder) error {
+		if len(resource) == 0 {
+			return fmt.Errorf("swarm resource cannot be empty")
+		}
+		b.swarmResourceEnvvars = []string{}
+		// if resource is a single string, split it by comma
+		if len(resource) == 1 && strings.Contains(resource[0], ",") {
+			candidates := strings.Split(resource[0], ",")
+			for _, c := range candidates {
+				trimmed := strings.TrimSpace(c)
+				if len(trimmed) > 0 {
+					b.swarmResourceEnvvars = append(b.swarmResourceEnvvars, trimmed)
+				}
+			}
+			return nil
+		}
+
+		b.swarmResourceEnvvars = append(b.swarmResourceEnvvars, resource...)
+		return nil
+	}
+}
+
+// WithPrivileged sets the privileged option for the CUDA image.
+// This is to allow testing the privileged mode of the container.
+// DO NOT USE THIS IN PRODUCTION CODE. FOR TESTING PURPOSES ONLY.
+func WithPrivileged(privileged bool) Option {
```

Comment:
Why is this not for "Production"? Even in production we need to determine whether a container is privileged.